### ISSUE TITLE
New version crashes on some Android phones

### ISSUE THREAD

**Issue by nohli** [florent37/Flutter-AssetsAudioPlayer#463]
**Flutter Version**

My version : 1.26.0-17.3.pre (beta)

**Lib Version**

My version : 2.0.13+5 (latest)

**Platform (Android / iOS / web) + version**

Platform : Android 8.0 & 8.1 (in Google Play Pre-Launch-Reports)

**Describe the bug**

```
FATAL EXCEPTION: main
Process: com.achimsapps.breathe, PID: 13322
android.app.RemoteServiceException: Bad notification posted from package com.achimsapps.breathe: Couldn't inflate contentViewsjava.lang.IllegalArgumentException: setShowActionsInCompactView: action 0 out of bounds (max -1)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1881)
	at android.os.Handler.dispatchMessage(Handler.java:105)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6938)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

```
FATAL EXCEPTION: main
Process: com.achimsapps.breathe, PID: 21047
android.app.RemoteServiceException: Bad notification posted from package com.achimsapps.breathe: Couldn't inflate contentViewsjava.lang.IllegalArgumentException: setShowActionsInCompactView: action 0 out of bounds (max -1)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1792)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6523)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:857)
```
This version works fine:

```
  assets_audio_player:
    git:
      url: https://github.com/florent37/Flutter-AssetsAudioPlayer.git
      ref: c454d02
```

This version (and the latest from pub.dev) crashes:

```
  assets_audio_player:
    git:
      url: https://github.com/florent37/Flutter-AssetsAudioPlayer.git
      ref: cba4cb3
```

So the causing change was probably made here: https://github.com/florent37/Flutter-AssetsAudioPlayer/compare/c454d02...cba4cb3

**Comment by kalismeras61** [comment]
Hey @nohli ,

Can yıu try with this commit ? : ddd9929e92c607e8fa6cafdbb02f461198a7b37d

**Comment by nohli** [comment]
Already building and deploying :)

**Comment by nohli** [comment]
@kalismeras61 crash seems fixed, thanks :)

Would make sense to publish this to pub.dev soon.

**Comment by kalismeras61** [comment]
@nohli  done. You can use new version on pub.dev

**Comment by nohli** [comment]
@kalismeras61 since the latest update I'm getting the error again in Pre-launch-reports:

```
android.app.RemoteServiceException: Bad notification posted from package com.achimsapps.breathe: Couldn't inflate contentViewsjava.lang.IllegalArgumentException: setShowActionsInCompactView: action 0 out of bounds (max -1)
android.app.RemoteServiceException: Bad notification posted from package com.achimsapps.breathe: Couldn't inflate contentViewsjava.lang.IllegalArgumentException: setShowActionsInCompactView: action 0 out of bounds (max -1)
```

**Comment by kalismeras61** [comment]
@nohli , can you give me more information, Device Manufacturer ? OS version ? etc.

**Comment by nohli** [comment]
Sure :)

```
Samsung Galaxy S9
1080x2009
Android 8.0 (SDK 26)
3,584 MB
ARM64_V8
ro_RO
```

```
FATAL EXCEPTION: main
Process: com.achimsapps.breathe, PID: 7591
android.app.RemoteServiceException: Bad notification posted from package com.achimsapps.breathe: Couldn't inflate contentViewsjava.lang.IllegalArgumentException: setShowActionsInCompactView: action 0 out of bounds (max -1)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1881)
	at android.os.Handler.dispatchMessage(Handler.java:105)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6938)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

---------------------

```
Nokia Nokia 1
480x854
Android 8.1 (SDK 27)
1,024 MB
ARM_V7
es_US
```

```
FATAL EXCEPTION: main
Process: com.achimsapps.breathe, PID: 18783
android.app.RemoteServiceException: Bad notification posted from package com.achimsapps.breathe: Couldn't inflate contentViewsjava.lang.IllegalArgumentException: setShowActionsInCompactView: action 0 out of bounds (max -1)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1792)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6523)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:857)
```

**Comment by nohli** [comment]
> Hey @nohli ,
> 
> Can you try with this commit ? : [ddd9929](https://github.com/florent37/Flutter-AssetsAudioPlayer/commit/ddd9929e92c607e8fa6cafdbb02f461198a7b37d)

Looks like this commit was reversed:
https://github.com/florent37/Flutter-AssetsAudioPlayer/blob/master/android/src/main/kotlin/com/github/florent37/assets_audio_player/notification/NotificationService.kt

**Comment by kalismeras61** [comment]
@nohli I know the issue, dod you remember old commit was working with lock screen control ?

**Comment by nohli** [comment]
Have controls disabled, so no. What's the issue?

**Comment by kalismeras61** [comment]
> Have controls disabled, so no. What's the issue?

Because of this. You disabled controls but the commit trying search actions (buttons) I think should revert back changes.

**Comment by nohli** [comment]
@kalismeras61 not sure if I understand what you mean.
Why did you revert the fixing commit mentioned above, was it on purpose?
And is it possible to add https://github.com/florent37/Flutter-AssetsAudioPlayer/commit/ddd9929e92c607e8fa6cafdbb02f461198a7b37d again without side-effects?

I think, it should be possible to show a sound notification with no enabled controls without the package crashing the app.

**Comment by kalismeras61** [comment]
Yes will do that.

**Comment by kalismeras61** [comment]
@nohli issue fixed please use 3.0.3

**Comment by nohli** [comment]
Thanks again!! 😊
